File: 2891-maximum-beauty-of-an-array-after-applying-operation/maximum-beauty-of-an-array-after-applying-operation.go
func maximumBeauty(nums []int, k int) (ans int) {
	mx := slices.Max(nums)
	diff := make([]int, mx+2)
	for _, x := range nums {
		diff[max(x-k, 0)]++
		diff[min(x+k+1, mx+1)]--
	}
	count := 0
	for _, x := range diff {
		count += x
		ans = max(ans, count)
	}
	return
}

File: 2891-maximum-beauty-of-an-array-after-applying-operation/maximum-beauty-of-an-array-after-applying-operation.cs
public class Solution {
    static int MaxInArray(int[] nums)
    {
        int mx = nums[0];
        foreach (int num in nums)
        {
            if (num > mx) mx = num;
        }
        return mx;
    }
    
    public int MaximumBeauty(int[] nums, int k) {
        int mx = MaxInArray(nums);
        int[] diff = new int[mx + 2];

        foreach (int x in nums)
        {
            diff[Math.Max(x - k, 0)]++;
            diff[Math.Min(x + k + 1, mx + 1)]--;
        }

        int count = 0, ans = 0;
        foreach (int x in diff)
        {
            count += x;
            ans = Math.Max(ans, count);
        }

        return ans;
    }
}